test-tests/shared/src/test/scala/zio/test/environment/ClockSpec.scala

```diff
@@ -0,0 +1,156 @@
+        },
+        testM("adjust correctly advances currentTime") {
+          for {
+            time1 <- currentTime(TimeUnit.NANOSECONDS)
+            _     <- adjust(1.millis)
+            time2 <- currentTime(TimeUnit.NANOSECONDS)
+          } yield assert(time2 - time1, equalTo(1.millis.toNanos))
+        },
+        testM("adjust correctly advances currentDateTime") {
+          for {
+            time1 <- currentDateTime
+            _     <- adjust(1.millis)
+            time2 <- currentDateTime
+          } yield assert((time2.toInstant.toEpochMilli - time1.toInstant.toEpochMilli), equalTo(1L))
+        },
+        testM("adjust does not produce sleeps") {
+          adjust(1.millis)
+          assertM(sleeps, isEmpty)
+        },
+        testM("setTime correctly sets nanotime") {
+          for {
+            _    <- setTime(1.millis)
+            time <- clock.nanoTime
+          } yield assert(time, equalTo(1.millis.toNanos))
+        },
+        testM("setTime correctly sets currentTime") {
+          for {
+            _    <- setTime(1.millis)
+            time <- currentTime(TimeUnit.NANOSECONDS)
+          } yield assert(time, equalTo(1.millis.toNanos))
+        },
+        testM("setTime correctly sets currentDateTime") {
+          for {
+            _    <- TestClock.setTime(1.millis)
+            time <- currentDateTime
+          } yield assert(time.toInstant.toEpochMilli, equalTo(1.millis.toMillis))
+        },
+        testM("setTime does not produce sleeps") {
+          for {
+            _      <- setTime(1.millis)
+            sleeps <- sleeps
+          } yield assert(sleeps, isEmpty)
+        },
+        testM("setTimeZone correctly sets timeZone") {
+          setTimeZone(ZoneId.of("UTC"))
+          assertM(timeZone, equalTo(ZoneId.of("UTC")))
+
+        },
+        testM("setTimeZone does not produce sleeps") {
+          setTimeZone(ZoneId.of("UTC"))
+          assertM(sleeps, isEmpty)
+        },
+        testM("timeout example from TestClock documentation works correctly") {
+          val example = for {
+            fiber  <- ZIO.sleep(5.minutes).timeout(1.minute).fork
+            _      <- TestClock.adjust(1.minute)
+            result <- fiber.join
+          } yield result == None
+          assertM(example, isTrue)
+        } @@ nonFlaky(100),
+        testM("recurrence example from TestClock documentation works correctly") {
+          val example = for {
+            q <- Queue.unbounded[Unit]
+            _ <- (q.offer(()).delay(60.minutes)).forever.fork
+            a <- q.poll.map(_.isEmpty)
+            _ <- TestClock.adjust(60.minutes)
+            b <- q.take.as(true)
+            c <- q.poll.map(_.isEmpty)
+            _ <- TestClock.adjust(60.minutes)
+            d <- q.take.as(true)
+            e <- q.poll.map(_.isEmpty)
+          } yield a && b && c && d && e
+          assertM(example, isTrue)
+        },
+        testM("fiber time is not subject to race conditions") { //times out
```

Comment:
The issue is that you are creating and adjusting your own `TestClock` but `sleep` is using the `TestClock` in the environment so is never waking up. Use the pattern above of accessing the `TestClock` in the environment and setting it to zero and I think this should work fine.

Reply:
@adamgfraser That makes sense. Fixed the timeout issue but created another :) fibertime seems to be 100x what it should be "200 did not satisfy equalTo(2)" (in millis)